crates/sui-node/src/lib.rs

```diff
@@ -602,20 +611,8 @@ impl SuiNode {
                     info!("This node is no longer a validator after reconfiguration");
                 }
             } else if self.state.is_validator() {
-                info!("Promoting the node from fullnode to validator, starting grpc server");
-
-                let (validator_server_handle, narwhal_manager, consensus_adapter) =
-                    Self::construct_validator_components(
-                        &self.config,
-                        self.state.clone(),
-                        self.checkpoint_store.clone(),
-                        self.state_sync.clone(),
-                        self.registry_service.clone(),
-                    )
-                    .await?;
-                self.validator_server_handle = Some(validator_server_handle);
-                self.narwhal_manager = Some(narwhal_manager);
-                self.consensus_adapter = Some(consensus_adapter);
+                info!("Switching from fullnode to validator. Must exit and restart");
+                return Ok(());
```

Comment:
What are the intended logistics of this? What if the operator restarts before reconfiguration, uploading the necessary configs ahead of time, in that case would there be an issue with upgrading the node here? Would it become burdensome to require operators to be available at the exact time when their node is elected to become a validator by the system to do manual restart?

Reply:
Addressed